lib/blocks.php

```diff
@@ -58,6 +58,80 @@ function gutenberg_parse_blocks( $content ) {
 }
 
 /**
+ * Given an array of parsed blocks, returns content string.
+ *
+ * @since 1.3.0
+ *
+ * @param array $blocks Parsed blocks.
+ *
+ * @return string Content string.
+ */
+function gutenberg_serialize_blocks( $blocks ) {
+	return implode( '', array_map( 'gutenberg_serialize_block', $blocks ) );
+}
+
+/**
+ * Given a parsed block, returns content string.
+ *
+ * @since 1.3.0
+ *
+ * @param array $block Parsed block.
+ *
+ * @return string Content string.
+ */
+function gutenberg_serialize_block( $block ) {
+	// Return content of unknown block verbatim.
+	if ( ! isset( $block['blockName'] ) ) {
+		return $block['rawContent'];
```

Comment:
quick ear-mark here as I think we're going to probably rename `rawContent` to something like `innerHtml`

Reply:
>quick ear-mark here as I think we're going to probably rename `rawContent` to something like `innerHtml`

I think that would be good. One idea I had originally was that, in parsing blocks, we could save effort in reserializing them since we should already know the original string text from which the block was parsed. Do you think it could be reasonable to still have something like `rawContent` returned from the parser which is assigned as the raw text of the block, comments and all?

Then again, there could be value in having the serializer available anyways, and not relying on blocks _maybe_ having this `rawContent` property available (e.g. not available if created from PHP, only if parsed from text).

Reply:
my original intention in bringing up `rawContent` was an _exception_ - that most blocks would return a tree of children but a few blocks which "need" the raw HTML would get it instead of the tree.

I can imagine use cases for the full body of the block but at the same time anticipate that providing as much is more of an enabler for working around the system than with it.

the way the PEG generator works this can be somewhat tricky to provide both - not sure why - which means that it may take more time if we want that to happen (we end up having to carry the `rawContent` attribute to each level of token and then reduce at the top level to stitch them together)